docs/kubernetes/operator/instrumenting-nodejs.md

```diff
@@ -0,0 +1,94 @@
+-->
+
+In this section you will learn how to:
+
+- Enable auto-instrumentation of a Node.js application following any of the supported methods, such as:
+  - Adding an annotation to the deployment pods.
+  - Adding an annotation to the namespace.
+- Verify that auto-instrumentation libraries are injected and configured correctly.
+- Confirm data is flowing to **Kibana Observability**.
+
+For demonstration purposes, we assume the application to be instrumented is a deployment named `nodejs-app` running in the `nodejs-ns` namespace.
+
+1. Ensure you have successfully [installed the OpenTelemetry Operator](./README.md), and confirm that the following `Instrumentation` object exists in the system:
+
+```bash
+$ kubectl get instrumentation -n opentelemetry-operator-system
+NAME                      AGE    ENDPOINT                                                                                                
+elastic-instrumentation   107s   http://opentelemetry-kube-stack-daemon-collector.opentelemetry-operator-system.svc.cluster.local:4318
+```
+> [!NOTE]
+> If your `Instrumentation` object has a different name or is created in a different namespace, you will have to adapt the annotation value in the next step.
+
+2. Enable auto-instrumentation of the Node.js application using one of the following methods:
+
+  - Edit the workload definition and include the annotation under `spec.template.metadata.annotations`:
+
+    ```yaml
+    spec:
+    ...
+      template:
+        metadata:
+          labels:
+            app: nodejs-app
+          annotations:
+            instrumentation.opentelemetry.io/inject-nodejs: opentelemetry-operator-system/elastic-instrumentation
+    ...
+    ```
+
+  - Alternatively, add the annotation at namespace level to apply auto-instrumentation in all Pods of the namespace:
+
+    ```bash
+    kubectl annotate namespace nodejs-ns instrumentation.opentelemetry.io/inject-nodejs=opentelemetry-operator-system/elastic-instrumentation
+    ```
+
+3. Restart application:
+
+  Once the annotation has been set, the Pods need to be recreated for the instrumentation libraries to be injected.
+
+    ```bash
+    kubectl rollout restart deployment nodejs-app -n nodejs
+    ```
+
+4. Verify the auto-instrumentation resources are injected in the Pod:
+
+  Node.js apps are instrumented by the OpenTelemetry Operator with the following actions:
+(TBD / WE NEED TO REVIEW HOW THIS IS DONE / NODE_OPTIONS env var?)
+
+  - ??It adds an init container in the Pod with the objective of copying the SDK to a shared volume.
+
+  - ??Defines an `emptyDir volume` mounted in both containers.
+
+  - ??Adds `NODE_OPTIONS` and other OTEL related environment variables.
+
+5. Confirm data is flowing through in **Kibana**:
+
+  - Open Observability -> Applications -> Service Inventory, and determine if:
+    - The application appears in the list of services.
+    - The application shows transactions and metrics.
```

Comment:
It is possible we want to hedge on there being transactions to see. If the deployed application has not received any requests/activity, then there won't be any trace data.

Reply:
Good point, I think I'm going to add in all guides (and the troubleshooting doc) something about it, like ensuring the application gets some traffic and generates some traces before checking in Kibana.

Also it's usually very easy to generate traffic using `kubectl port-forward,  so we can add an example.

What would you suggest?

Reply:
I have used a few things in the past:
- Specifically add a request-to-self in the application code so that there is always at least one request to it (perhaps a little awkward to have that in the code)
- Rely on traffic from a Docker Compose healthcheck. However, we obviously aren't using Compose here. I'm ignorant enough with k8s to know if here is some typicaly/stock healthcheck facility.
- Explicitly have a loadgen container somewhere in the system.  This can be nice for showing application metrics. However, adding a loadgen container perhaps complicates the "getting started" demo.

Reply:
I have added a `TIP` on that section and pinged you and Mike for review. Marking this as resolved for now.